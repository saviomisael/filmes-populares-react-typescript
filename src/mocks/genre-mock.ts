export const genreMock = [{ id: 1, name: 'Action' }];
