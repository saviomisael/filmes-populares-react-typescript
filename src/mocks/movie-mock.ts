export const movieMock = {
  id: 1,
  name: 'Duna',
  rating: 9,
  releaseDate: new Date(),
  poster: '/s9E9W77HS8zEQvsrpz5aEUTKnvD.jpg',
  overview:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.',
};
