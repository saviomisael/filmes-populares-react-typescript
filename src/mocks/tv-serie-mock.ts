export const tvSerieMock = { id: 1, name: 'Round 6', rating: 7.5, poster: '/6gcHdboppvplmBWxvROc96NJnmm.jpg' };
