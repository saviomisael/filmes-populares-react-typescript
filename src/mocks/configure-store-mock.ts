import { configureStore } from '@reduxjs/toolkit';
import { tvSerieMockReducer } from './tv-serie-duck-mock';

export const storeMock = configureStore({
  reducer: {
    tvSeries: tvSerieMockReducer,
  },
});
