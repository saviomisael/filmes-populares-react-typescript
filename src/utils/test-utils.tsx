import { render } from '@testing-library/react';
import { MemoryRouter, Switch, Route } from 'react-router-dom';
import { BaseStyles } from 'styles/base';
import { ThemeProvider } from 'styled-components';
import { theme } from 'styles/theme';
import { ReactNode } from 'react';
import { Provider } from 'react-redux';
import { storeMock } from 'mocks/configure-store-mock';

type optionsWithRouter = {
  initialRoutes?: string[];
  pathname: string;
};

export const renderTheme = (children: ReactNode) =>
  render(
    <ThemeProvider theme={theme}>
      <BaseStyles />
      {children}
    </ThemeProvider>,
  );

export const renderWithStoreProvider = (children: ReactNode) =>
  render(
    <Provider store={storeMock}>
      <ThemeProvider theme={theme}>
        <BaseStyles />
        {children}
      </ThemeProvider>
    </Provider>,
  );

export const renderWithRouter = (children: ReactNode, options: optionsWithRouter) => {
  const initialRoutes = options.initialRoutes && options.initialRoutes.length > 0 ? options.initialRoutes : ['/'];

  return render(
    <Provider store={storeMock}>
      <ThemeProvider theme={theme}>
        <MemoryRouter initialEntries={initialRoutes}>
          <Switch>
            <Route path={options.pathname} render={() => children} />
          </Switch>
        </MemoryRouter>
      </ThemeProvider>
    </Provider>,
  );
};
