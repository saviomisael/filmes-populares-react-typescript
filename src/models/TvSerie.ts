import { IMAGE_URL } from 'utils/constants';

export class TvSerie {
  constructor(
    public readonly id: number,
    public readonly name: string,
    public readonly rating: number,
    public readonly poster: string,
  ) {
    this.poster = `${IMAGE_URL}${poster}`;
  }
}
