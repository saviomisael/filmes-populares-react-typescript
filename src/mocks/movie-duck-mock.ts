import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { Movie } from 'models/Movie';
import { MovieState } from 'store/ducks/movie-duck';
import { genresMock } from './genre-mock';
import { movieListMock, movieMock } from './movie-mock';

const initialState: MovieState = {
  movies: movieListMock.map((x) => new Movie(x.id, x.name, x.rating, x.releaseDate, x.poster, x.overview)),
  singleMovie: new Movie(
    movieMock.id,
    movieMock.name,
    movieMock.rating,
    movieMock.releaseDate,
    movieMock.poster,
    movieMock.overview,
    [...genresMock],
  ),
};

const movieSlice = createSlice({
  name: 'movie',
  initialState,
  reducers: {
    addMovies: (state, action: PayloadAction<Movie[]>) => {
      state.movies = action.payload;
    },
    updateSingleMovie: (state, action: PayloadAction<Movie>) => {
      state.singleMovie = action.payload;
    },
  },
});

export const { actions: movieMockActions, reducer: movieMockReducer } = movieSlice;
