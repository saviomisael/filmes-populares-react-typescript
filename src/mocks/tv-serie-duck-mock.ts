import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { TvSerie } from 'models/TvSerie';
import { tvSerieListMock } from 'mocks/tv-serie-mock';

export interface TvSerieState {
  tvSeries: TvSerie[];
}

const initialState: TvSerieState = {
  tvSeries: tvSerieListMock.map((x) => new TvSerie(x.id, x.name, x.rating, x.poster)),
};

const tvSerieSlice = createSlice({
  name: 'tvserie',
  initialState,
  reducers: {
    addSeries: (state, action: PayloadAction<TvSerie[]>) => {
      state.tvSeries = action.payload;
    },
  },
});

export const { actions: tvSerieMockActions, reducer: tvSerieMockReducer } = tvSerieSlice;
