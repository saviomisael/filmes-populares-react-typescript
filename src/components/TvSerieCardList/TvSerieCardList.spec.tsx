import { screen } from '@testing-library/react';
import { renderTheme } from 'utils/test-utils';
import { TvSerieCardList } from '.';
import { tvSerieListMock } from 'mocks/tv-serie-mock';
import { TvSerie } from 'models/TvSerie';

describe('<TvSerieCardList />', () => {
  const series = tvSerieListMock.map((x) => new TvSerie(x.id, x.name, x.rating, x.poster));

  it('should render many list item elements', () => {
    renderTheme(<TvSerieCardList series={series} />);

    const liElements = screen.getAllByRole('listitem');

    expect(liElements).toHaveLength(5);
  });

  it('should matches with snapshot', () => {
    renderTheme(<TvSerieCardList series={series} />);

    const container = screen.getByTestId('tvserielistcontainer');

    expect(container).toMatchSnapshot();
  });
});
