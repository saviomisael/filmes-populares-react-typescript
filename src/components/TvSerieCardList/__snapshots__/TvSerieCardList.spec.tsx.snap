// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TvSerieCardList /> should matches with snapshot 1`] = `
.c6 {
  font-weight: 400;
  font-size: 1.8rem;
  color: #FFFFFF;
  text-align: center;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-column-gap: 1.4rem;
  column-gap: 1.4rem;
  color: #FFFFFF;
}

.c7::before {
  content: url(star.svg);
}

.c1 {
  width: 24rem;
}

.c5 {
  width: 100%;
  position: absolute;
  right: 0;
  bottom: 0.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1.5rem 0;
  visibility: hidden;
  opacity: 0;
  -webkit-transition: visibility 0.3s,opacity 0.3s;
  transition: visibility 0.3s,opacity 0.3s;
  border-bottom-left-radius: inherit;
  border-bottom-right-radius: inherit;
  background: #413B3B88;
}

.c2 {
  border-radius: 1.2rem;
  position: relative;
}

.c2:hover .c4 {
  visibility: visible;
  opacity: 1;
}

.c3 {
  width: 100%;
  border-radius: inherit;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

<ul
  class="c0"
  data-testid="tvserielistcontainer"
>
  <li>
    <article
      class="c1"
    >
      <figure
        class="c2"
      >
        <img
          alt="poster da série Round 6"
          class="c3"
          src="https://image.tmdb.org/t/p/original/6gcHdboppvplmBWxvROc96NJnmm.jpg"
        />
        <figcaption
          class="c4 c5"
        >
          <p
            class="c6"
          >
            Round 6
          </p>
          <h4
            class="c7"
          >
            7.5
          </h4>
        </figcaption>
      </figure>
    </article>
  </li>
  <li>
    <article
      class="c1"
    >
      <figure
        class="c2"
      >
        <img
          alt="poster da série Round 6"
          class="c3"
          src="https://image.tmdb.org/t/p/original/6gcHdboppvplmBWxvROc96NJnmm.jpg"
        />
        <figcaption
          class="c4 c5"
        >
          <p
            class="c6"
          >
            Round 6
          </p>
          <h4
            class="c7"
          >
            7.5
          </h4>
        </figcaption>
      </figure>
    </article>
  </li>
  <li>
    <article
      class="c1"
    >
      <figure
        class="c2"
      >
        <img
          alt="poster da série Round 6"
          class="c3"
          src="https://image.tmdb.org/t/p/original/6gcHdboppvplmBWxvROc96NJnmm.jpg"
        />
        <figcaption
          class="c4 c5"
        >
          <p
            class="c6"
          >
            Round 6
          </p>
          <h4
            class="c7"
          >
            7.5
          </h4>
        </figcaption>
      </figure>
    </article>
  </li>
  <li>
    <article
      class="c1"
    >
      <figure
        class="c2"
      >
        <img
          alt="poster da série Round 6"
          class="c3"
          src="https://image.tmdb.org/t/p/original/6gcHdboppvplmBWxvROc96NJnmm.jpg"
        />
        <figcaption
          class="c4 c5"
        >
          <p
            class="c6"
          >
            Round 6
          </p>
          <h4
            class="c7"
          >
            7.5
          </h4>
        </figcaption>
      </figure>
    </article>
  </li>
  <li>
    <article
      class="c1"
    >
      <figure
        class="c2"
      >
        <img
          alt="poster da série Round 6"
          class="c3"
          src="https://image.tmdb.org/t/p/original/6gcHdboppvplmBWxvROc96NJnmm.jpg"
        />
        <figcaption
          class="c4 c5"
        >
          <p
            class="c6"
          >
            Round 6
          </p>
          <h4
            class="c7"
          >
            7.5
          </h4>
        </figcaption>
      </figure>
    </article>
  </li>
</ul>
`;
